pyiron_contrib/RDM/storagejob.py

```diff
@@ -0,0 +1,324 @@
+        """
+        super().__init__(init=init, table_name=table_name)
+        self._available_storage_types = ['local', 's3']
+        self._storage_type = None
+        self.storage_type = storage_type
+
+    def create_group(self, name):
+        """
+        Add a new empty subcontainer under the given key.
+
+        Args:
+            name (str): key under which to store the new subcontainer in this container
+
+        Returns:
+            DataContainer: the newly created subcontainer
+
+        >>> pl = DataContainer({})
+        >>> pl.create_group("group_name")
+        DataContainer([])
+        >>> list(pl.group_name)
+        []
+        """
+        self[name] = DataContainer()
+        return self[name]
+
+    @classmethod
+    def _read_only_error(cls):
+        raise RuntimeError("The storage type cannot be changed when it is already in use.")
+
+    @property
+    def storage_type(self):
+        return self._storage_type
+
+    @storage_type.setter
+    def storage_type(self, storage_type):
+        if storage_type not in self._available_storage_types:
+            raise ValueError(f"Expected 'storage_type' to be one of {self._available_storage_types} "
+                             f"but got {storage_type}.")
+        self._storage_type = storage_type
+        for storage in self._available_storage_types:
+            self[storage] = False
+        self[storage_type] = True
+
+
+class StorageJob(GenericJob):
+    """Job to store files associated with meta data either locally, or at a remote data service like s3."""
+
+    def __init__(self, project, job_name):
+
+        super().__init__(project, job_name)
+        self.server.run_mode.interactive = True
+        self._stored_files = DataContainer(table_name='stored_files')
+        self._storage_type = StorageType(storage_type='local', table_name="storage_type")
+        self._external_storage = None
+
+    def _before_generic_remove_child(self):
+        if self._storage_type.s3:
+            _remove_s3_working_directory(self._external_storage, self.path)
+
+    def check_setup(self):
+        """
+        this function is called in self._run_if_new() if self.server.run_mode.queue
+        i.e. self.server.run_mode = queue if one sets a queue
+        afterwards self.save() and self.run() is called again -> self.run_if_created  followed by (in this case always) 
+        self.run_if_scheduler 
+        run_if_scheduler is the function called to sent the job to the cluster, however, it already sends the 
+        stuff to the queue which is obviously not intended if this is used as ssh storage system.
+        TODO: 
+        - overwrite run_if_scheduler to not trigger a job to be run on the queing system. 
+        - make some file_handler class to allow for 'store/receive/delete file' with a uniform interface for
+            - local
+            - s3
+            - ssh
+        """
+        raise NotImplementedError("Storing files remotely is not yet supported.")
```

Comment:
I am A-OK with this for this PR, but this docstring really feels like it needs to be in our high-level developer docs somewhere...!!!